### DISCUSSION TITLE
Cache remote jwks in cloudflare workers

### DISCUSSION THREAD

**Discussion by kansson** [panva/jose#661] [Ideas]
I'm looking for an option to specify the `cf` options when fetching remote jwks in cloudflare workers as the current caching solution isn't effective here. https://developers.cloudflare.com/workers/examples/cache-using-fetch/

**Comment by panva** [comment]
Since there would be no mechanism for invalidating that `cf:` cache coming from the library you're better off making the cached fetch yourself and piping its Response's parsed json to `createLocalJWKSet`. You'd re-fetch yourself when https://github.com/panva/jose/blob/adc0f69362d4c7c95bc0dde9b871f0f95c599a49/docs/classes/util_errors.JWKSNoMatchingKey.md is thrown

**Reply by kansson** [reply]
That was easy thank you. There is also the cache api which can be invalidated. https://developers.cloudflare.com/workers/runtime-apis/cache/

**Reply by panva** [reply]
It's unlikely I would build platform-specific fetch cache invalidation in anyway.
